content/blog/2025/10/2025-10-10-post-mortem.md

```diff
@@ -0,0 +1,182 @@
+
+We’re considering changing the background colour of the terminal dynamically depending on the role the database is playing in the cluster. An idea we floated is to monitor the presence of the `standby.signal` file in the database data directory to know whether it is a primary or a secondary database, and update the terminal’s background colour accordingly. This is not a silver bullet since the background colour would only change after a command has been sent, but that would already be an improvement.
+
+We also discussed wrapper scripts around sensitive commands (such as an alias for `rm`) or automating some operations, such as starting a base backup from primary to secondary as a means to minimise risk. 
+
+We could restore the service after 24h offline, *without any data loss* despite losing both our primary and secondary databases. This accounts for a great Recovery Point Objective and is testament to our PITR processes that we test regularly. We should take pride in the recovery, but we need to work on a shorter Recovery Time Objective, we’re currently talking to service providers to get free infrastructure that would make it easier and faster to recover.
+
+### We can have better tools
+
+We upgraded wal-g on all servers, not just the Disaster Recovery Server, and have done a round of Disaster Recovery testing with it. We didn’t explore yet how we can ensure the Disaster Recovery Server and the production servers can stay aligned.
+
+At the next hardware refresh, we will explore using ZFS so we can make local snapshots and recover much more quickly from not so happy accidents such as accidentally wiping the wrong database.
+
+### We have a great community and providers
+
+We received a lot of support on social media where we communicated actively around the incident. This was welcomed positively by the broad community, despite our status page not receiving the attention it deserved. We’re adding steps to our incident response playbook to update [status.matrix.org](http://status.matrix.org) as the canonical source of truth during an incident, and liaise with the advocacy team to keep social media updated as well.
+
+The SRE team would like to thank our hosting provider Mythic Beasts. They reached out quickly and proactively when adding new disks, reporting the errors they were seeing. They have been much more than just a pair of remote hands. They also reached out with an offer of support during the incident.
+
+Finally, we’d like to sincerely apologise again to everyone impacted by the outage. We hope you found the post-mortem informative and you’d like to talk about it more, several of us will be at the [Matrix Conference 2025](https://conference.matrix.org) in Strasbourg. In addition to a flurry of great talks, there will be workshops about how to set up a Matrix homeserver and tune the clients to your liking\!
```

Comment:
outdated

Reply:
consider pointing at the conference wrap-up or fosdem instead!

Reply:
I think we can leave it out, since the wrap-up will be published in the following days